system/Model.php

```diff
@@ -612,13 +581,34 @@ public function insert($data, bool $returnID = true)
 	 * Updates a single record in $this->table. If an object is provided,
 	 * it will attempt to convert it into an array.
 	 *
+	 * updates a single item by id
+	 * $model->update(id, data)
+	 *
+	 * updates multiple items by array of ids
+	 * $model->update(array_id, data)
+	 *
 	 * @param int|string   $id
 	 * @param array|object $data
 	 *
 	 * @return bool
 	 */
-	public function update($id, $data)
+	public function update($id, $data = null)
 	{
+        // because we're going to use "where" for the items to updated
+        // id became a surrogate for data argument
+        // not ideal, but lets see where this goes...
+        if ($data === null)
+        {
+            $data = $id;
+            $id = null;
+        }
+
+        // you can not proceed.
+        if ($id === null && $data === null) return false;
```

Comment:
I'd prefer to throw an exception here. It seems like any time this would happen would be something the developer should have caught so we should fail loudly so they have a chance to fix it. Just returning false here would make the bug harder to find and easier to slip through into production.

Reply:
Yeah, that's a good idea, otherwise, it's a silent death with no reason why.